# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [4]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [25]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2         V3        V4        V5        V6        V7       
1 0.7881533 0.09067125 0.8411875 0.8928760 0.4147388 0.2268053 0.3683460
2 0.9006603 0.77644796 0.1868457 0.5313053 0.4311195 0.5239130 0.4767423
  V8        V9        V10       V11       V12       V13         V14      
1 0.3757095 0.1939247 0.6554509 0.1909811 0.5897435 0.797131109 0.9826911
2 0.4103410 0.7043355 0.2824360 0.5982016 0.2925924 0.009760606 0.5742338
  V15        V16       V17        V18       V19       V20       
1 0.03096807 0.4754822 0.14799662 0.7669558 0.1558947 0.04035054
2 0.37579670 0.5015291 0.07773861 0.6902871 0.6474650 0.13087097

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [27]:
df2 <- df1 %>%
  gather()

head(df2)

key value     
1 V1  0.78815331
2 V1  0.90066031
3 V2  0.09067125
4 V2  0.77644796
5 V3  0.84118751
6 V3  0.18684570

## Assignment 2

We will generate some data:

In [20]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [21]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [23]:
df5 <- full_join(df3,df4, by = 'id')

df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook